src/confirmations_impl.h

```diff
@@ -2,98 +2,199 @@
+  std::vector<std::string> signed_blinded_payment_token_json_bundles;
+  std::vector<std::string> fully_submitted_payment_bundles;
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  std::string real_wallet_address_;
+  std::string real_wallet_address_secret_key_;
+  std::string local_server_confirmation_key_;
+  std::vector<std::string> local_original_confirmation_tokens_;
+  std::vector<std::string> local_blinded_confirmation_tokens_;
+  std::string confirmation_id_;
+  std::string local_original_payment_token_;
+  std::string local_blinded_payment_token_;
+  std::string blinded_payment_token_;
+  base::DictionaryValue* map_;
+  std::string bundle_json_;
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  void SetConfirmationsStatus();
+
+  std::string GetServerUrl();
+  int GetServerPort();
+
+  void Step1StoreTheServersConfirmationsPublicKeyAndGenerator(
+      std::string confirmations_GH_pair,
+      std::vector<std::string> bat_names,
+      std::vector<std::string> bat_keys);
+
+  void Step2RefillConfirmationsIfNecessary(
+      std::string real_wallet_address,
+      std::string real_wallet_address_secret_key,
+      std::string server_confirmation_key);
+  void Step2bRefillConfirmationsIfNecessary(
+      const std::string& url,
+      const int response_status_code,
+      const std::string& response,
+      const std::map<std::string, std::string>& headers);
+  void Step2cRefillConfirmationsIfNecessary(
+      const std::string& url,
+      const int response_status_code,
+      const std::string& response,
+      const std::map<std::string, std::string>& headers);
+  void OnStep2RefillConfirmationsIfNecessary(const Result result);
+
+  void Step3RedeemConfirmation(std::string real_creative_instance_id);
+  void Step3bRedeemConfirmation(
+      const std::string& url,
+      const int response_status_code,
+      const std::string& response,
+      const std::map<std::string, std::string>& headers);
+  void OnStep3RedeemConfirmation(const Result result);
+
+  void Step4RetrievePaymentIOUs();
+
+  void Step5CashInPaymentIOUs(std::string real_wallet_address);
+  void Step5bCashInPaymentIOUs(
+      const std::string& url,
+      const int response_status_code,
+      const std::string& response,
+      const std::map<std::string, std::string>& headers);
+  void OnStep5CashInPaymentIOUs(const Result result);
+
+  bool VerifyBatchDLEQProof(
+      std::string proof_string,
+      std::vector<std::string> blind_strings,
+      std::vector<std::string> signed_strings,
+      std::string public_key_string);
+
+  void PopFrontConfirmation();
+  void PopFrontPayment();
+
+  void SaveState();
+  void OnStateSaved(const Result result);
+  void LoadState();
+  void OnStateLoaded(const Result result, const std::string& json);
+  void ResetState();
+  void OnStateReset(const Result result);
+
+  std::string ToJSON();
+  bool FromJSON(std::string json);
```

Comment:
Should be const params. We should probably check all of the function signatures when fixing this.

Reply:
This library is not const correct, but I did not want to change original code, so ok to fix later?